### ISSUE TITLE
A way to generate/rewrite directives to pin all the things for a stable, reproducible build

### ISSUE THREAD

**Issue by armanbilge** [VirtusLab/scala-cli#2145]
**Is your feature request related to a problem? Please describe.**

Scala CLI sets many values implicitly, such as Scala Version, Scala.js / Native Version, etc. Additionally, dependencies may be added with the `latest.release` version. While this is excellent for development and rapid prototyping, eventually you may want a stable, reproducible build when you deploy into production.

**Describe the solution you'd like**

A way for Scala CLI to provide or rewrite all the relevant directives to get a stable, reproducible build.

**Describe alternatives you've considered**

You can attempt to do this manually, but it's not always obvious all the things that are necessary to pin, and it takes some effort to track down the versions.

For example there is a [`--js-cli-version`](https://scala-cli.virtuslab.org/docs/reference/cli-options/#--js-cli-version) option that must also be pinned if you want stability/reproducibility, but this is marked as an "internal" option.

**Additional context**

See also https://github.com/scala/toolkit/issues/20#issuecomment-1560731598.

**Comment by tgodzik** [comment]
Thanks for reporting! I think this would be nice to have, but we don't have currently the capacity to work on it.

I think this should be an additional command that we could group all the potential fixes maybe together with scalafix (could it even be a scalafix rule)?

Something like:
`scala fix [rules]`

Rules could be the scalafix ones or ones provided by default.

**Comment by bjaglin** [comment]
> I think this should be an additional command that we could group all the potential fixes maybe together with scalafix (could it even be a scalafix rule)?
> 
> Something like: `scala fix [rules]`

Agreed. Related discussion about scala-cli-centric fixes as scalafix rules: https://github.com/VirtusLab/scala-cli/pull/3400#issuecomment-2628524125.
